File: Free Html To Pdf Converter for ASP.NET MVC/[VB]-Free Html To Pdf Converter for ASP.NET MVC/VB/SelectPdf.Samples/Controllers/PdfConverterViewerPreferencesController.vb
Imports SelectPdf

Namespace Controllers
    Public Class PdfConverterViewerPreferencesController
        Inherits Controller

        ' GET: PdfConverterViewerPreferences
        Public Function Index() As ActionResult
            Return View()
        End Function

        <HttpPost> _
        Public Function SubmitAction(collection As FormCollection) As ActionResult
            ' read parameters from the webpage
            Dim url As String = collection("TxtUrl")

            Dim page_layout As String = collection("DdlPageLayout")
            Dim pageLayout As PdfViewerPageLayout = DirectCast(
                [Enum].Parse(GetType(PdfViewerPageLayout), page_layout, True),
                PdfViewerPageLayout)

            Dim page_mode As String = collection("DdlPageMode")
            Dim pageMode As PdfViewerPageMode = DirectCast(
                [Enum].Parse(GetType(PdfViewerPageMode), page_mode, True),
                PdfViewerPageMode)

            Dim centerWindow As Boolean = collection("ChkCenterWindow") = "on"
            Dim displayDocTitle As Boolean = collection("ChkDisplayDocTitle") = "on"
            Dim fitWindow As Boolean = collection("ChkFitWindow") = "on"
            Dim hideMenuBar As Boolean = collection("ChkHideMenuBar") = "on"
            Dim hideToolbar As Boolean = collection("ChkHideToolbar") = "on"
            Dim hideWindowUI As Boolean = collection("ChkHideWindowUI") = "on"

            ' instantiate a html to pdf converter object
            Dim converter As New HtmlToPdf()

            ' set converter options
            converter.Options.ViewerPreferences.CenterWindow = centerWindow
            converter.Options.ViewerPreferences.DisplayDocTitle = displayDocTitle
            converter.Options.ViewerPreferences.FitWindow = fitWindow
            converter.Options.ViewerPreferences.HideMenuBar = hideMenuBar
            converter.Options.ViewerPreferences.HideToolbar = hideToolbar
            converter.Options.ViewerPreferences.HideWindowUI = hideWindowUI

            converter.Options.ViewerPreferences.PageLayout = pageLayout
            converter.Options.ViewerPreferences.PageMode = pageMode
            converter.Options.ViewerPreferences.NonFullScreenPageMode =
                PdfViewerFullScreenExitMode.UseNone

            ' create a new pdf document converting an url
            Dim doc As PdfDocument = converter.ConvertUrl(url)

            ' save pdf document
            Dim pdf As Byte() = doc.Save()

            ' close pdf document
            doc.Close()

            ' return resulted pdf document
            Dim fileResult As FileResult = New FileContentResult(pdf, "application/pdf")
            fileResult.FileDownloadName = "Document.pdf"
            Return fileResult
        End Function
    End Class
End Namespace

File: Free Html To Pdf Converter for ASP.NET MVC/[C#]-Free Html To Pdf Converter for ASP.NET MVC/C#/SelectPdf.Samples/Controllers/PdfConverterViewerPreferencesController.cs
using System;
using System.Web.Mvc;

namespace SelectPdf.Samples.Controllers
{
    public class PdfConverterViewerPreferencesController : Controller
    {
        // GET: PdfConverterViewerPreferences
        public ActionResult Index()
        {
            return View();
        }

        [HttpPost]
        public ActionResult SubmitAction(FormCollection collection)
        {
            // read parameters from the webpage
            string url = collection["TxtUrl"];

            string page_layout = collection["DdlPageLayout"];
            PdfViewerPageLayout pageLayout = (PdfViewerPageLayout)Enum.Parse(
                typeof(PdfViewerPageLayout), page_layout, true);

            string page_mode = collection["DdlPageMode"];
            PdfViewerPageMode pageMode = (PdfViewerPageMode)Enum.Parse(
                typeof(PdfViewerPageMode), page_mode, true);

            bool centerWindow = collection["ChkCenterWindow"] == "on";
            bool displayDocTitle = collection["ChkDisplayDocTitle"] == "on";
            bool fitWindow = collection["ChkFitWindow"] == "on";
            bool hideMenuBar = collection["ChkHideMenuBar"] == "on";
            bool hideToolbar = collection["ChkHideToolbar"] == "on";
            bool hideWindowUI = collection["ChkHideWindowUI"] == "on";

            // instantiate a html to pdf converter object
            HtmlToPdf converter = new HtmlToPdf();

            // set converter options
            converter.Options.ViewerPreferences.CenterWindow = centerWindow;
            converter.Options.ViewerPreferences.DisplayDocTitle = displayDocTitle;
            converter.Options.ViewerPreferences.FitWindow = fitWindow;
            converter.Options.ViewerPreferences.HideMenuBar = hideMenuBar;
            converter.Options.ViewerPreferences.HideToolbar = hideToolbar;
            converter.Options.ViewerPreferences.HideWindowUI = hideWindowUI;

            converter.Options.ViewerPreferences.PageLayout = pageLayout;
            converter.Options.ViewerPreferences.PageMode = pageMode;
            converter.Options.ViewerPreferences.NonFullScreenPageMode =
                PdfViewerFullScreenExitMode.UseNone;

            // create a new pdf document converting an url
            PdfDocument doc = converter.ConvertUrl(url);

            // save pdf document
            byte[] pdf = doc.Save();

            // close pdf document
            doc.Close();

            // return resulted pdf document
            FileResult fileResult = new FileContentResult(pdf, "application/pdf");
            fileResult.FileDownloadName = "Document.pdf";
            return fileResult;
        }
    }
}